docs/advanced/custom-components.md

```diff
@@ -1,84 +1,92 @@
-
-    const [x, y] = this._context.viewport.project([longitude, latitude]);
-
-    const markerStyle = {
-      position: 'absolute',
-      background: '#fff',
-      left: x,
-      top: y
-    };
-
-    return (
-      <div ref={this._containerRef}
-        style={markerStyle} >
-        ({longitude}, {latitude})
-      </div>
-    );
-  }
+import MapGL, {MapContext} from 'react-map-gl';
+
+function CustomMarker(props) {
+  const context = React.useContext(MapContext);
+  
+  const {longitude, latitude} = props;
+
+  const [x, y] = context.viewport.project([longitude, latitude]);
+
+  const markerStyle = {
+    position: 'absolute',
+    background: '#fff',
+    left: x,
+    top: y
+  };
+
+  return (
+    <div style={markerStyle} >
+      ({longitude}, {latitude})
+    </div>
+  );
 }
-```
-
-## Properties
-
-The following properties are handled by the `BaseControl` component:
-
-##### `captureScroll` {Boolean} - default: `false`
-Stop propagation of mouse wheel event to the map component. Can be used to stop map from zooming when this component is scrolled.
-
-##### `captureDrag` {Boolean} - default: `true`
-Stop propagation of dragstart event to the map component. Can be used to stop map from panning when this component is dragged.
-
-##### `captureClick` {Boolean} - default: `true`
-Stop propagation of click event to the map component. Can be used to stop map from calling the `onClick` callback when this component is clicked.
-
-##### `captureDoubleClick` {Boolean} - default: `true`
-Stop propagation of dblclick event to the map component. Can be used to stop map from zooming when this component is double clicked.
-
-##### `capturePointerMove` {Boolean} - default: `false`
-Stop propagation of pointermove event to the map component. Can be used to stop map from calling the `onMouseMove` or `onTouchMove` callback when this component is hovered.
-
-## Private Members
-
-##### `_containerRef`
 
-A React [ref](https://reactjs.org/docs/refs-and-the-dom.html#creating-refs) object.
-
-Should be assigned to the `ref` prop of the root DOM element of this component. Required to leverage the `capture*` props.
-
-##### `_context`
+function App() {
+  return (
+    <MapGL longitude={-122.45} latitude={37.78} zoom={12} mapboxApiAccessToken={MAPBOX_TOKEN}>
+      <CustomeMarker longitude={-122.45} latitude={37.78} />
+    </MapGL>
+  )
+}
+```
 
-An object containing the following fields:
+## Using the useMapControl hook
 
-- `viewport` {WebMercatorViewport} - the current viewport
-- `map` {mapboxgl.Map} - the Mapbox map instance
-- `eventManager` {EventManager} - the event manager. Only available if using `InteractiveMap`.
-- `isDragging` {Bool} - whether the map is being dragged. Only available if using `InteractiveMap`.
+`useMapControl` is an experimental API that makes it easier to create controls with event handling.
 
+```js
+import * as React from 'react';
+import MapGL, {_useMapControl as useMapControl} from 'react-map-gl';
```

Comment:
i noticed in here: https://github.com/visgl/react-map-gl/pull/1278/files#diff-b8e38018e9d59b4598ff7ff7f767e98641d1148372578b5323fbc916e02f67c9R22
we import directly `useMapControl`. Why do we import `_ useMapControl` in this statement?

Reply:
Which file are you referring to?